src/radix_connect/interaction_id.rs

```diff
@@ -0,0 +1,28 @@
+use crate::prelude::*;
+
+uniffi::custom_newtype!(WalletInteractionId, String);
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct WalletInteractionId(pub String);
+
+impl From<String> for WalletInteractionId {
+    fn from(value: String) -> Self {
+        Self(value)
+    }
+}
+
+impl From<&str> for WalletInteractionId {
```

Comment:
Ah ok, so best do this:

```rust
impl WalletInteractionId {
    pub fn new(id: impl AsRef<str>) -> Self {
        Self(id.as_ref().to_owned())
    }
}
```

That way we can pass both `&str` and `String` and we *avoid* `From<&str>`, since it is awfully close to `FromStr` which is throwing.